tasks/main.yml

```diff
@@ -627,6 +627,193 @@
         __mssql_conf_setting: "network forceencryption"
         __mssql_conf_setting_value: "{{ '1' if mssql_tls_enable else 'unset' }}"
 
+- name: Configure Active Directory authentication
+  vars:
+    __mssql_kinit_user: >-
+      {{ ad_integration_user }}@{{ ad_integration_realm
+      | upper }}
+  when: mssql_ad_configure | bool
+  block:
+    - name: Join to {{ ad_integration_realm }} realm
+      vars:
+        ad_integration_realm: "{{ ad_integration_realm }}"
+        ad_integration_password: "{{ ad_integration_password }}"
+      include_role:
+        name: linux-system-roles.ad_integration
+
+    - name: Ensure the adutil package
+      package:
+        name: adutil
+        state: present
+
+    - name: List valid kerberos tickets
+      shell: klist || true
+      register: __mssql_klist
+      ignore_errors: true
+      changed_when: false
+
+    - name: kinit as the {{ ad_integration_user }} priviledged user
+      shell: >-
+        echo {{ ad_integration_password | quote }}
+        | kinit {{ __mssql_kinit_user }}
+      when: >-
+        __mssql_kinit_user not in __mssql_klist.stdout
+      no_log: true
+      changed_when: false
+
+    - name: Check if {{ mssql_ad_sql_username }} SQL user exists
+      shell: >-
+        adutil account exists
+        --name {{ mssql_ad_sql_username }}
+        --accept-eula || true
```

Comment:
Are you using `shell` and `|| true` here to avoid having the ugly ERROR message in the log when using `ignore_errors: true`?

Reply:
Maybe there is a way to use `command` then parse `__mssql_adutil_account_exists` and use `failed_when` instead?

Reply:
> > Are you using `shell` and `|| true` here to avoid having the ugly ERROR message in the log when using `ignore_errors: true`?
> 
> Yes this was my idea. I remember system roles had some issues with these errors so I am trying to avoid this. Also, in the playbook output there are big red errorred tasks and single blue line underneath saying that this error is ignored, so this might confuse users too.

Hmm - maybe try using `failed_when: some condition` or even `failed_when: false`

Reply:
Done that, thank you